src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -131,16 +132,17 @@ public void periodic() {
         getRotation2d(),
         getPositions()
         );
-
-    for (int i=0; i<m_photonCameras.getFieldRelativePoseEstimators().size(); i++) {
-      Optional<EstimatedRobotPose> robotPoseEstimator = m_photonCameras.getFieldRelativePoseEstimators().get(i);
+    
+    ArrayList<Optional<EstimatedRobotPose>> photonRobotPoseList = m_photonCameras.getFieldRelativePoseEstimators();
+    for (int i=0; i<photonRobotPoseList.size(); i++) {
```

Comment:
This could be written as:

```java
photonRobotPoseList.forEach(robotPoseEstimator -> {
  [...]
});
```

Or, as long as the stuff in front of `forEach` isn't too complicated, and you don't need `photonRobotPoseList` for multiple operations:

```java
m_photonCameras.getFieldRelativePoseEstimators().foreach(robotPoseEstimator -> {
  [...]
});
```

It's usually a good idea to use "for each"-style iteration rather than indexed iteration whenever you don't explicitly need the index inside the loop, because there are fewer ways to mess up the boilerplate code.